python/vineyard/core/client.py

```diff
@@ -60,18 +62,25 @@ def __init__(
         port: Union[int, str] = None,
         # move host after port to make sure unnamed (host, port) works
         host: str = None,
-        endpoint: Tuple[str, Union[str, int]] = None,
+        endpoint: Union(str, Tuple[str, Union[str, int]]) = None,
```

Comment:
For endpoint that host and port are combined into a single string, it should be recongnized as `socket`.